azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PointToSiteConfiguration.java

```diff
@@ -0,0 +1,151 @@
+             * @return the next stage of the definition
+             */
+            WithRootCertificate<ParentT> withAzureCertificate();
+
+            WithAttach<ParentT> withRadiusAuthentication(String serverIPAddress, String serverSecret);
+        }
+
+        interface WithRootCertificate<ParentT> {
+            WithAttach<ParentT> withRootCertificate(String name, String certificateData);
+        }
+
+        interface WithRevokedCertificate<ParentT> {
+            WithAttach<ParentT> withRevokedCertificate(String name, String thumbprint);
+        }
+
+        /**
+         * The stage of a point-to-site configuration definition allowing to specify which tunnel type will be used.
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithTunnelType<ParentT> {
+            WithAttach<ParentT> withSstpOnly();
+
+            WithAttach<ParentT> withIkeV2Only();
+        }
+
+        /** The final stage of the point-to-site configuration definition.
+         * <p>
+         * At this stage, any remaining optional settings can be specified, or the point-to-site configuration definition
+         * can be attached to the parent virtual network gateway definition.
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithAttach<ParentT> extends
+                Attachable.InDefinition<ParentT>,
+                WithTunnelType<ParentT>,
+                WithRootCertificate<ParentT>,
+                WithRevokedCertificate<ParentT> {
+        }
+    }
+
+    /** The entirety of a point-to-site configuration definition.
+     * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+     */
+    interface Definition<ParentT> extends
+            DefinitionStages.Blank<ParentT>,
+            DefinitionStages.WithAuthenticationType<ParentT>,
+            DefinitionStages.WithAddressPool<ParentT>,
+            DefinitionStages.WithAttach<ParentT> {
+    }
+
+    /**
+     * Grouping of point-to-site configuration update stages.
+     */
+    interface UpdateStages {
+        interface WithAddressPool<ParentT> {
+            WithAuthenticationType<ParentT> withAddressPool(String addressPool);
+        }
+
+        /**
+         * The stage of the point-to-site configuration definition allowing to specify authentication type.
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithAuthenticationType<ParentT> {
+            /**
+             * Specifies that Azure certificate authentication type will be used.
+             * @return the next stage of the definition
+             */
+            Update withAzureCertificate();
+
+            Update withRadiusAuthentication(String serverIPAddress, String serverSecret);
```

Comment:
Can you "revoke" the RADIUS authentication once was set? Also what happens to the Azure native certification authorization if it was previously set and now you update with a RADIUS one?